### PR TITLE
Update macOS export page.

### PR DESCRIPTION
**Pull request by bruvzg**
Adds some notes to include information about hidden options and exporting from non macOS systems.

Fixes [#63100](https://github.com/godotengine/godot/issues/63100)

### CODE REVIEW

File: tutorials/export/exporting_for_macos.rst [hunk 1]
```diff
@@ -10,13 +10,15 @@ Exporting for macOS
...
+This bundle can be exported as is, packed in a ZIP archive or DMG disk image (only supported when exporting from a computer running macOS).
 
 Requirements
 ------------
 
--  To enable code signing and notarization, you must export from a computer running macOS with Xcode command line tools installed.
+-  To enable code signing with Apple Developer ID and notarization, you must export from a computer running macOS with Xcode command line tools installed.
+-  Ad-hoc code signing is supported on all platforms, without additional tools.
 -  Download the Godot export templates. Use the Godot menu: ``Editor > Manage Export Templates``.
+-  Valid and unique ``Bundle identifier`` should be set in the ``Application`` section of the export options.
```
Comment (mhilbrunner) [inline tutorials/export/exporting_for_macos.rst:21]:
```suggestion
-  A valid and unique ``Bundle identifier`` should be set in the ``Application`` section of the export options.
```

File: tutorials/export/exporting_for_macos.rst [hunk 2]
```diff
@@ -69,7 +73,10 @@ Signing Options
 
     To notarize an app, you must enable the ``Hardened Runtime`` and ``Timestamp``.
 
+    ``Hardened Runtime`` and ``Timestamp`` options are incompatible with ad-hoc signature and will be ignored.
```
Comment (mhilbrunner) [inline tutorials/export/exporting_for_macos.rst:76]:
```suggestion
    The ``Hardened Runtime`` and ``Timestamp`` options are incompatible with ad-hoc signing and will be ignored.
```

**Review by mhilbrunner — COMMENTED** [review]
Looks good, very small syntax nitpicks:

**Comment by bruvzg** [comment]
> small syntax nitpicks:

Fixed.

**Comment by mhilbrunner** [comment]
Merged, thank you! :)

**Comment by mhilbrunner** [comment]
Cherry-picked to 3.5.

**Pull request merged**